BEP7.md

```diff
@@ -0,0 +1,320 @@
+
+| **Field** | **Type** | **Description**              |
+| :--------- | :-------- | :---------------------------- |
+| Input     | []Input  | A set of transaction inputs  |
+| Output    | []Output | A set of transaction outputs |
+
+**Input Data Structure:**
+
+| **Field** | **Type** | **Description**                                              |
+| :--------- | :-------- | :------------------------------------------------------------ |
+| Address   | Address  | Address for token holders                                    |
+| Coins     | []Coin   | A set of sorted coins, one per currency. The symbols of coins are in descending order. |
+
+**Output Data Structure:**
+
+| **Field** | **Type** | **Description**                                              |
+| :--------- | :-------- | :------------------------------------------------------------ |
+| Address   | Address  | Address for token holders                                    |
+| Coins     | []Coin   | A set of sorted coins, one per currency. The denominations of coins are in descending order. |
+
+**Coin Structure:**
+
+| **Field** | **Type** | **Description**                                                 |
+| :--------- | :-------- | :------------------------------------------------------------ |
+| Denom     | string   | The symbol of a collection                                      |
+| ID        | string   | The id of a token in the collection                             |
+| Amount    | int64    | The amount is positive and can have a maximum of 8 digits of decimal and is boosted by 1e8 in order to store as int64. |
+
+**Transfer Process:**
+
+- Transferer initiators sign a transfer transaction and make it broadcasted to one of Binance Chain nodes
+- The Binance Chain node will check this transaction. If there is no error, then this transaction will be broadcasted to other Binance Chain nodes
+- Transfer transaction is committed on the blockchain by block proposer
+- Validators will verify the constraints on balance. The transfer tokens and fee will be deducted from the address of the transaction initiators.
+- Add the tokens to the destination addresses
+
+#### 6.2.3 Freeze Tokens
+
+A Binance Chain user could freeze some amount of tokens in his own address. The freeze transaction will lock his fund, thus this portion of tokens could not be used for the transactions, such as: creating orders, transferring to another account, paying fees and etc.  
+
+**Data Structure** **for Freeze Operation**: A data structure is needed to represent the freeze operation
+
+| **Field** | **Type** | **Description**                                                 |
+| :--------- | :-------- | :------------------------------------------------------------ |
+| Denom     | string   | The symbol of a collection - e.g. NNB-B90                       |
+| ID        | string   | The id of a token in the collection                             |
```

Comment:
Do you think the freeze period is necessary? If there is no freeze period, the owners can freeze and unfreeze their assets at anytime, thus the freeze operation will have no limitation for owners.